fenix/app/src/test/java/org/mozilla/fenix/library/history/state/HistoryNavigationMiddlewareTest.kt

```diff
@@ -0,0 +1,67 @@
+package org.mozilla.fenix.library.history.state
+
+import androidx.navigation.NavController
+import mozilla.components.support.test.any
+import mozilla.components.support.test.ext.joinBlocking
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.rule.MainCoroutineRule
+import mozilla.components.support.test.rule.runTestOnMain
+import mozilla.components.support.test.whenever
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.verify
+import org.mozilla.fenix.library.history.History
+import org.mozilla.fenix.library.history.HistoryFragmentAction
+import org.mozilla.fenix.library.history.HistoryFragmentState
+import org.mozilla.fenix.library.history.HistoryFragmentStore
+import org.mozilla.fenix.library.history.HistoryItemTimeGroup
+
+class HistoryNavigationMiddlewareTest {
+    @get:Rule
+    val coroutinesTestRule = MainCoroutineRule()
+
+    @Test
+    fun `WHEN regular history item clicked THEN item is opened in browser`() = runTestOnMain {
+        var openedInBrowser = false
+        val url = "url"
+        val history = History.Regular(0, "title", url, 0, HistoryItemTimeGroup.timeGroupForTimestamp(0))
+        val middleware = HistoryNavigationMiddleware(
+            navController = mock(),
+            openToBrowser = { item ->
+                if (item.url == url) {
+                    openedInBrowser = true
+                }
+            },
+            scope = this,
+        )
+        val store =
+            HistoryFragmentStore(HistoryFragmentState.initial, middleware = listOf(middleware))
+
+        store.dispatch(HistoryFragmentAction.HistoryItemClicked(history)).joinBlocking()
+
+        assertTrue(openedInBrowser)
+    }
+
+    @Test
+    fun `WHEN group history item clicked THEN navigate to history metadata fragment`() = runTestOnMain {
+        val title = "title"
+        val history = History.Group(0, title, 0, HistoryItemTimeGroup.timeGroupForTimestamp(0), listOf())
+        val navController = mock<NavController>()
+        whenever(navController.navigate(directions = any(), navOptions = any())).thenAnswer { }
+        val middleware = HistoryNavigationMiddleware(
+            navController = navController,
+            openToBrowser = { },
+            scope = this,
+        )
+        val store =
+            HistoryFragmentStore(HistoryFragmentState.initial, middleware = listOf(middleware))
+
+        store.dispatch(HistoryFragmentAction.HistoryItemClicked(history)).joinBlocking()
+
+        verify(navController).navigate(
+            directions = any(),
+            navOptions = any(),
+        )
```

Comment:
based on your test title, should this be actually verifying that its navigating to the metadata fragment?

Reply:
I ran into some issues using mockito to verify the right call with this, see [thread](https://mozilla.slack.com/archives/G01EXNFFAVC/p1689094086224469?thread_ts=1689093900.869579&cid=G01EXNFFAVC) for details. I didn't want to spin my wheels too hard on it now, but I could file a follow-up if you'd like. IMO this method is still preferable to capturing the `navController.navigate` call in a lamdba, passing that through to the middleware, and asserting that the lambda is called under test. This way the test at least captures that the navController is being interacted with, while the lambda solution would not. 

I can change the test name if it seems to mismatch, though!

Reply:
I was more inquiring about the usage of `any`, but I'm cool with your approach since you've already tried to do it the "right" way, and I agree that doing the lambda check doesn't get us much more compared to this.